actionview/test/template/tag_helper_test.rb

```diff
@@ -150,27 +284,48 @@ def test_tag_honors_html_safe_with_escaped_array_class
     assert_equal '<p class="song> play&gt;" />', str
   end
 
+  def test_tag_builder_honors_html_safe_with_escaped_array_class
+    str = tag.p(:class => ['song>', raw('play>')])
+    assert_equal '<p class="song&gt; play>"></p>', str
+
+    str = tag.p(:class => [raw('song>'), 'play>'])
+    assert_equal '<p class="song> play&gt;"></p>', str
+  end
+
   def test_skip_invalid_escaped_attributes
     ['&1;', '&#1dfa3;', '& #123;'].each do |escaped|
       assert_equal %(<a href="#{escaped.gsub(/&/, '&amp;')}" />), tag('a', :href => escaped)
+      assert_equal %(<a href="#{escaped.gsub(/&/, '&amp;')}"></a>), tag.a(:href => escaped)
     end
   end
 
   def test_disable_escaping
     assert_equal '<a href="&amp;" />', tag('a', { :href => '&amp;' }, false, false)
   end
 
+  def test_tag_builder_disable_escaping
+    assert_equal '<a href="&amp;"></a>', tag.a({ :href => '&amp;' }, false)
+    assert_equal '<a href="&amp;">cnt</a>', tag.a({ :href => '&amp;' }, false) { "cnt"}    
+    assert_equal '<br data-hidden="&amp;">', tag.br({ "data-hidden": '&amp;' }, false)  
+    assert_equal '<a href="&amp;">content</a>', tag.a("content", { :href => '&amp;' }, false)
+    assert_equal '<a href="&amp;">content</a>', tag.a({ :href => '&amp;' }, false) { "content"}
```

Comment:
Last param `escape` is always boolean, when set to false disables attribute value escaping. Ported from old tag helper. i.e.
tag.img({src: "open &amp; shut.png"}, false)
tag.p('<script>evil_js</script>', {}, false)